[RFR] Play the 15-puzzle in a webbrowser

Luwangel opened the pull request:
- [x] Add guzzle, twig and webprofiler as dependencies
- [x] Use the last version of the 15-puzzle 
- [x] Build a game module to communicate with the go webserver
- [x] Create a template to render the puzzle

Luwangel committed:
Add guzzle as dependency

Luwangel committed:
Add twig as dependency

Luwangel committed:
Add a web profiler as dev dependency

Luwangel committed:
Synchronize the 15 puzzle go project and update the readme

fzaninotto commented:
Can you add a screenshot as soon as you have something visible?

Luwangel committed:
Use csa guzzle bundle instead of guzzle

Luwangel committed:
Create game and game serializer classes to manage the game through the api

Luwangel committed:
Remove the hello route

Luwangel committed:
Register the puzzle api as a service and create a specific class to manage it

jdemangeon commented on app/config/packages/config.yaml:
```diff
@@ -0,0 +1,6 @@
+  clients:
+    puzzle_api:
+      config:
+        base_uri: http://15-puzzle-go:2000
```
Please use parameters to set this url

jdemangeon commented on app/src/Api/GameSerializer.php:
```diff
@@ -0,0 +1,29 @@
+use App\Game\Game;
+
+class GameSerializer {
+  private $_serializer;
```
s/$_serializer/$serializer/

jdemangeon commented on app/src/Api/GameSerializer.php:
```diff
@@ -0,0 +1,29 @@
+
+use App\Game\Game;
+
+class GameSerializer {
```
You can implement the SerializeInterface to "fix" class implementation
http://api.symfony.com/3.1/Symfony/Component/Serializer/SerializerInterface.html

jdemangeon replied:
You can implement the SerializerInterface to "fix" class implementation
http://api.symfony.com/3.1/Symfony/Component/Serializer/SerializerInterface.html

sedy-bot committed:
Typo fix s/$_serializer/$serializer/

As requested by jdemangeon at https://github.com/marmelab/15-puzzle-web/pull/3#discussion_r149083308

Luwangel committed:
Add asset as dependency

Luwangel committed:
Display a shuffled puzzle

jdemangeon commented on app/src/Controller/GameController.php:
```diff
@@ -0,0 +1,26 @@
+  }
+ 
+  public function start() {
+    $game = $this->api->new(4);
```
Please use a constant for 4 (to know what is it)

jdemangeon replied:
Can you use a linter? There's an help here https://github.com/djoos/Symfony-coding-standard

jdemangeon commented on app/tests/Controller/MainControllerTest.php:
```diff
@@ -0,0 +1,19 @@
+
+class MainControllerTest extends WebTestCase {
+    public function testStart() {
+
```
Useless line

alexisjanvier commented on app/templates/main.html.twig:
```diff
@@ -0,0 +1,13 @@
+
+{% block content %}
+  <div class="section">
+    <a class="waves-effect waves-light btn" href="/game"><i class="material-icons left">add_circle_outline</i>Start a new game</a>
```
`waves-effect`, `waves-light`, you're selling dreams... :)

Luwangel committed:
Use the serializer service and fix the game api

Luwangel committed:
Use an environment variable to store the go api

Luwangel committed:
Use the route move-tile instead of the route move

Luwangel commented:
Main page:
![capture du 2017-11-07 10-29-05](https://user-images.githubusercontent.com/5584839/32486313-921b2818-c3a6-11e7-9ba3-e45c51599f2e.png)

Game page:
![capture du 2017-11-07 10-28-36](https://user-images.githubusercontent.com/5584839/32486311-91d3714e-c3a6-11e7-9f0f-0d01a743ad2d.png)

alexisjanvier committed:
Fix Useless line
